### PR TITLE
MdePkg/DxeRngLib: Refactoring and improvements

### PR DESCRIPTION
**Pull request by pierregondois**
# Description

Improve the DxeRngLib so that:
- the RNG prototocol is located only once instead of at each
  random number generation
- algorithms that are not available on a platform are not called
  when trying to generate a random number
- gEfiRngAlgorithmArmRndr support is added
- PcdEnforceSecureRngAlgorithms can be set to allow/avoid non-secure
  algorithms

- [ ] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.

## How This Was Tested

Tested on a Juno. Associated patches to be provided.

### CODE REVIEW

**Comment by pierregondois** [comment]
Associated `edk2-platforms` PR: https://github.com/tianocore/edk2-platforms/pull/185

File: MdePkg/MdePkg.dsc [hunk 1]
```diff
@@ -34,6 +34,7 @@
 
 [LibraryClasses]
   SafeIntLib|MdePkg/Library/BaseSafeIntLib/BaseSafeIntLib.inf
+  TimerLib|MdePkg/Library/BaseTimerLibNullTemplate/BaseTimerLibNullTemplate.inf
```
Comment (mdkinney) [inline MdePkg/MdePkg.dsc:37]: How is this related to these changes?  I do not see any new usage of TimerLib

Comment (pierregondois) [reply MdePkg/MdePkg.dsc:37]: Hello Michael,
the patch is actually not needed, it solved a dependency I introduced in a local patch ...

**Review by kraxel — APPROVED** [review]

File: MdePkg/Library/DxeRngLib/DxeRngLib.c [hunk 2]
```diff
@@ -1,17 +1,129 @@
...
+        if (mFirstAlgo == MAX_UINTN) {
+          mFirstAlgo = Index1;
+        }
+
+        break;
+      }
+    }
+  }
+
+exit_handler:
```
Comment (lgao4) [inline MdePkg/Library/DxeRngLib/DxeRngLib.c:129]: Please use upper case for Lable name

Comment (samimujawar) [reply MdePkg/Library/DxeRngLib/DxeRngLib.c:129]: Hi Liming,

Comment (lgao4) [reply MdePkg/Library/DxeRngLib/DxeRngLib.c:129]: Yes. Labe name should be same to Variable name. It can be ExitHandler.

**Review by lgao4 — APPROVED** [review]

**Pull request merged**